src/test/java/uk/gov/pay/api/it/PaymentRefundsResourceITest.java

```diff
@@ -0,0 +1,183 @@
+                .body("_links.payment.href", is(paymentLocationFor(CHARGE_ID)));
+    }
+
+    @Test
+    public void getRefundById_shouldGetNonAuthorized_whenPublicAuthRespondsUnauthorised() {
+
+        publicAuthMock.respondUnauthorised();
+
+        getPaymentRefundByIdResponse(API_KEY, CHARGE_ID, REFUND_ID)
+                .statusCode(401);
+    }
+
+    @Test
+    public void getRefundById_shouldReturnNotFound_whenRefundDoesNotExist() {
+
+        publicAuthMock.mapBearerTokenToAccountId(API_KEY, GATEWAY_ACCOUNT_ID);
+        connectorMock.respondRefundNotFound(GATEWAY_ACCOUNT_ID, CHARGE_ID, "unknown-refund-id");
+
+        getPaymentRefundByIdResponse(API_KEY, CHARGE_ID, REFUND_ID)
+                .statusCode(404)
+                .contentType(JSON)
+                .body("code", is("P0700"))
+                .body("description", is("Not found"));
+    }
+
+    @Test
+    public void getRefundById_returns500_whenConnectorRespondsWithResponseOtherThan200Or404() {
+
+        publicAuthMock.mapBearerTokenToAccountId(API_KEY, GATEWAY_ACCOUNT_ID);
+        connectorMock.respondRefundWithError(GATEWAY_ACCOUNT_ID, CHARGE_ID, REFUND_ID);
+
+        getPaymentRefundByIdResponse(API_KEY, CHARGE_ID, REFUND_ID)
+                .statusCode(500)
+                .contentType(JSON)
+                .body("code", is("P0798"))
+                .body("description", is("Downstream system error"));
+    }
+
+    @Test
+    public void getRefunds_shouldGetValidResponse() {
+
+        publicAuthMock.mapBearerTokenToAccountId(API_KEY, GATEWAY_ACCOUNT_ID);
+
+        PaymentRefundJsonFixture refund_1 = new PaymentRefundJsonFixture(100L, CREATED_DATE, "100", "available", new ArrayList<>());
+        PaymentRefundJsonFixture refund_2 = new PaymentRefundJsonFixture(300L, CREATED_DATE, "300", "available", new ArrayList<>());
+
+        connectorMock.respondWithGetAllRefunds(GATEWAY_ACCOUNT_ID, CHARGE_ID, refund_1, refund_2);
+
+        getPaymentRefundsResponse(API_KEY, CHARGE_ID)
+                .statusCode(200)
+                .contentType(JSON)
+                .body("payment_id", is(CHARGE_ID))
+                .body("_links.self.href", is(paymentRefundsLocationFor(CHARGE_ID)))
+                .body("_links.payment.href", is(paymentLocationFor(CHARGE_ID)))
+                .body("_embedded.refunds.size()", is(2))
+                .body("_embedded.refunds[0].refund_id", is("100"))
+                .body("_embedded.refunds[0].created_date", is(CREATED_DATE))
+                .body("_embedded.refunds[0].amount", is(100))
+                .body("_embedded.refunds[0].status", is("available"))
+                .body("_embedded.refunds[0]._links.size()", is(2))
+                .body("_embedded.refunds[0]._links.self.href", is(paymentRefundLocationFor(CHARGE_ID, "100")))
+                .body("_embedded.refunds[0]._links.payment.href", is(paymentLocationFor(CHARGE_ID)));
+    }
```

Comment:
idea of the test was it would break if the structure of refund changes, but can add the extra checks if it helps